src/Qir/Runtime/prerequisites.ps1

```diff
@@ -11,7 +11,7 @@ if ($Env:ENABLE_QIRRUNTIME -ne "false") {
         if (!(Get-Command clang        -ErrorAction SilentlyContinue) -or `
             !(Get-Command clang-format -ErrorAction SilentlyContinue) -or `
             (Test-Path Env:/AGENT_OS)) {
-            choco install llvm --version=14.0.6 --allow-downgrade
+            choco install llvm --version=15.0.3 --allow-downgrade
```

Comment:
I believe the upgrade of LLVM/clang requires a separate set of PRs (in more than one repo in parallel).
See the example of the [previous migration](https://github.com/microsoft/qsharp-runtime/pull/1065). 
In the description there is a reference to a parallel PR. 
Among the files changed there are documentation changes, a number of changes in the build scripts, in the C++ source files (because the updated clang-format has different requirements).

I would recommend to break up this PR into 3 - 5 smaller PRs.